extension/natives/navmesh.h

```diff
@@ -62,6 +62,119 @@ NAVMESHNATIVE(CollectSurroundingAreas)
 	return CREATEHANDLE(SurroundingAreasCollector, pCollector);
 }
 
+class SMPathCost : public IPathCost
+{
+public:
+	SMPathCost(IPluginFunction *pFunc) : m_pFunc(pFunc)
+	{
+	}
+
+	virtual float operator()(CNavArea *area, CNavArea *fromArea, const CNavLadder *ladder, const CFuncElevator *elevator, float length) const override final
+	{
+		if (m_pFunc && m_pFunc->IsRunnable())
```

Comment:
I'm quite unfamiliar with `IPluginFunction` but is the call to `IsRunnable` necessary?

Reply:
I believe so. It's used a few times in SM in similar cases like here:
https://github.com/alliedmodders/sourcemod/blob/1fbe5e1daaee9ba44164078fe7f59d862786e612/core/logic/smn_database.cpp#L242
https://github.com/alliedmodders/sourcemod/blob/1fbe5e1daaee9ba44164078fe7f59d862786e612/core/logic/smn_timers.cpp#L137

Reply:
Good enough for me!